server/src/main/java/org/elasticsearch/search/SearchFeatures.java

```diff
@@ -21,6 +22,10 @@ public final class SearchFeatures implements FeatureSpecification {
 
     @Override
     public Set<NodeFeature> getFeatures() {
-        return Set.of(KnnVectorQueryBuilder.K_PARAM_SUPPORTED, LUCENE_10_0_0_UPGRADE);
+        return Set.of(
+            KnnVectorQueryBuilder.K_PARAM_SUPPORTED,
+            LUCENE_10_0_0_UPGRADE,
+            MatchQueryBuilder.MATCH_QUERY_SEMANTIC_TEXT_SUPPORTED
```

Comment:
I think its something that should be checked by the rewriter as you don't want the coordinator to rewrite to a semantic query and not all nodes support it or something.

Reply:
I moved this to `InferenceFeatures`, is that what you were thinking?

Reply:
@kderusso the purpose of these features is to verify that the cluster fully satisfies them. If we are not actually using them for anything other than tests, we don't need it at all.

But, I am saying maybe the match rewrite should fail if that feature isn't fully supported in the current cluster.

Reply:
Thanks. I think we might be able to get away with this being a test feature. Thinking it through, if the rewrite isn't supported it will fail anyway, since the fields don't match? WDYT?

Reply:
Moved to test features